src/main/java/ca/wolfram/beta/symath/NodeType.java

```diff
@@ -5,12 +5,15 @@
 /**
  * Created by Allan Wang on 2017-05-06.
  * <p>
- * {@link #argCount} maximum number of arguments for type; -1 for inf
+ * {@link #argCount}
+ * key for number of arguments
+ * exactly x if x >= 0
+ * at least |x| if x < 0
  */
 public enum NodeType {
 
     CONSTANT(0), MATH_CONSTANT(0), VARIABLE(0),
-    ADD(-1), NEGATE(1), MULTIPLY(-1), POWER(2);
+    ADD(-2), SUBTRACT(-2), NEGATE(1), MULTIPLY(-2), DIVIDE(-2), POWER(2);
```

Comment:
I am still not sure about adding these 2 lines. because they aren't actual nodes